### PR TITLE
Build android cmake

### PR DESCRIPTION
**Pull request by AndistormEm**

### CODE REVIEW

File: buildme [hunk 1]
```diff
@@ -1,53 +1,30 @@
...
-ARCH=$(uname -m)
-ARM64=OFF
-CMAKE_TOOLCHAIN_FILE=../makefiles/cmake/toolchains/arm-linux-gnueabihf.cmake
 
-if [ "$1" = "--debug" ]; then
-	BUILDTYPE=Debug
-	shift
-fi
+CMAKE_BUILD_TYPE=Release
+CMAKE_TOOLCHAIN_FILE=/home/andreas/Code/Android/SDK/ndk/21.3.6528147/build/cmake/android.toolchain.cmake
```
Comment (JanKehren) [inline buildme:4]: @AndistormEm This shouldn't be hardcoded right ?

File: buildme [hunk 2]
```diff
@@ -1,53 +1,30 @@
...
-	CMAKE_TOOLCHAIN_FILE=../makefiles/cmake/toolchains/aarch64-linux-gnu.cmake
-	shift
-fi
+ANDROID_ABI=armeabi-v7a
+ANDROID_TOOLCHAIN=clang
+ANDROID_ARM_NEON=ON
+ANDROID_NATIVE_API_LEVEL=24
+ANDROID_NO_UNDEFINED=OFF # when "ON" show all undefined symbols as linker errors
+ANDROID_SO_UNDEFINED=OFF # when "ON" allow undefined symbols in shared libraries
+DESTDIR=../build2
```
Comment (JanKehren) [inline buildme]: Is there a reason why we call the dir `build2` instead of `build` ?

File: buildme [hunk 3]
```diff
@@ -1,53 +1,31 @@
...
 
-if [ "$1" = "--aarch64" ]; then
-	ARM64=ON
-	CMAKE_TOOLCHAIN_FILE=../../../makefiles/cmake/toolchains/aarch64-linux-gnu.cmake
-	shift
-fi
+ANDROID_ABI=armeabi-v7a
+ANDROID_TOOLCHAIN=clang
+ANDROID_ARM_NEON=ON
+ANDROID_NATIVE_API_LEVEL=24
```
Comment (JanKehren) [inline buildme:9]: Does this mean we have to make different prebuilds for different Android versions ? Android 7.1 has API 25 and Android 9 uses API 28 why do we use 24 here ?

File: buildme (hunk 2 above)
Comment (JanKehren) [reply buildme]: sorry saw that you changed this

File: buildme (hunk 1 above)
Comment (AndistormEm) [reply buildme:4]: I adopted it from our old build script, I don't think everyone has the ndk at the same place, and I don't know anything about an entry in the environment path.

File: buildme (hunk 3 above)
Comment (AndistormEm) [reply buildme:9]: I adopt this parameter from our old build script, too. I found this:

Specifies the minimum API level supported by the application or library. This value corresponds to the application's minSdkVersion. [See also](https://developer.android.com/ndk/guides/cmake#android_platform).

So I guess a smaller value shouldn't make it bader.

**Pull request merged**